### PR TITLE
Set default Torch version if YY.MM format is not met

### PR DESCRIPTION
**Pull request by thomasdhc**
# What does this PR do ?

Set default nvidia torch version 

**Collection**: [nlp]

# Changelog 
- Update nvidia torch version if YY.MM formatting is not met

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_base_model.py [hunk 1]
```diff
@@ -379,8 +379,11 @@ def _enable_nvidia_optimizations(self):
         # NVIDIA container version check
         nvidia_torch_version = os.getenv('NVIDIA_PYTORCH_VERSION', None)
 
-        # Support DLFW master container
-        if nvidia_torch_version == 'master':
+        def is_official_release_version(nvidia_torch_version):
+            return re.fullmatch("[0-9][0-9]\.[0-9][0-9].*", nvidia_torch_version)  # "YY.MM.*"
```
Comment (ericharper) [inline nemo/collections/nlp/models/language_modeling/megatron_base_model.py:383]: This is currently in the nemo 1.0 path, we'll need something similar for 2.0

**Comment by minitu** [comment]
LGTM

File: nemo/collections/nlp/models/language_modeling/megatron_base_model.py (hunk 1 above)
Comment (gautham-kollu) [reply nemo/collections/nlp/models/language_modeling/megatron_base_model.py:383]: @thomasdhc Can we unblock this for 1.0 and file a JIRA for the 2.0 code path ?

**Review by gautham-kollu — APPROVED** [review]
Approving.

Please file a corresponding bug for 2.0.

**Pull request merged**